src/ploigos_step_runner/step_implementers/generate_metadata/dotnet.py

```diff
@@ -0,0 +1,145 @@
+                                                                  If None / empty string will not auto increment version.
+`auto-increment-all-module-versions` | No        | True         | If auto incrementing version, auto increment version in all sub modules if True.
+
+Result Artifacts
+----------------
+Results artifacts output by this step.
+
+Result Artifact Key                     | Description
+----------------------------------------|------------
+`app-version`                           | Value to use for `version` portion of semantic version \
+                                          (https://semver.org/). Uses the version read out of the given pom file.
+
+`dotnet-auto-increment-version-output` | Standard out and standard error from running maven to auto increment version.
+"""# pylint: disable=line-too-long
+
+import os.path
+import xml.etree.ElementTree as ET
+from ploigos_step_runner.results import StepResult
+from ploigos_step_runner.step_implementer import StepImplementer
+
+
+
+
+DEFAULT_CONFIG = {
+    'csproj-file': 'dotnet-app.csproj'
+}
+
+REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS = [
+    'csproj-file'
+]
+
+
+class Dotnet(StepImplementer):
+    """`StepImplementer` for the `generate-metadata` step using Dotnet.
+    """
+
+    @staticmethod
+    def step_implementer_config_defaults():
+        """Getter for the StepImplementer's configuration defaults.
+
+        Returns
+        -------
+        dict
+            Default values to use for step configuration values.
+
+        Notes
+        -----
+        These are the lowest precedence configuration values.
+        """
+        return DEFAULT_CONFIG
+
+    @staticmethod
+    def _required_config_or_result_keys():
+        """Getter for step configuration or previous step result artifacts that are required before
+        running this step.
+
+        See Also
+        --------
+        _validate_required_config_or_previous_step_result_artifact_keys
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys or previous step result artifacts
+            that are required before running the step.
+        """
+        return REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS
+
+    def _validate_required_config_or_previous_step_result_artifact_keys(self):
+        """Validates that the required configuration keys or previous step result artifacts
+        are set and have valid values.
+
+        Validates that:
+        * required configuration is given
+        * given 'csproj-file' exists
+
+        Raises
+        ------
+        AssertionError
+            If step configuration or previous step result artifacts have invalid required values
+        """
+        super()._validate_required_config_or_previous_step_result_artifact_keys()
+
+        csproj_file = self.get_value('csproj-file')
+        assert os.path.exists(csproj_file), f'Given csproj file (csproj-file) does not exist: {csproj_file}'
```

Comment:
Move this into _run_step() or a method called by _run_step() and then delete the surrounding method